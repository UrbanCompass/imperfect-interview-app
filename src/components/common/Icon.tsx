import React from 'react';
import styled from '@emotion/styled';
import classnames from 'classnames';
import getCxIcon from '@uc/cx-icons';

interface IconProps extends React.HTMLAttributes<HTMLElement>, IconWrapperProps {
  className?: string;
  left?: boolean;
  right?: boolean;
  name: string;
}

interface IconWrapperProps {
  size?: string;
}

const I = styled.i<IconWrapperProps>`
  width: ${({ size }: IconWrapperProps): string | undefined => size};
  height: ${({ size }: IconWrapperProps): string | undefined => size};
  svg {
    height: ${({ size }: IconWrapperProps): string | undefined => size};
    width: ${({ size }: IconWrapperProps): string | undefined => size};
  }
`;

export function Icon(props: IconProps): JSX.Element {
  const { left, right, name, className, ...rest } = props;
  const classNames = classnames(
    'cx-icon',
    'cx-btn-icon',
    { 'cx-btn-icon--left': left },
    { 'cx-btn-icon--right': right },
    className
  );

  const icon = getCxIcon(name);

  return <I dangerouslySetInnerHTML={{ __html: icon }} className={classNames} {...rest} />;
}

export function ExitIcon(props): JSX.Element {
  return <Icon name="cx-icon-x_16x16" {...props} />;
}

export function EllipsisIcon(props): JSX.Element {
  return <Icon name="cx-icon-verticalEllipsis_16x16" {...props} />;
}

export function TrashIcon(props): JSX.Element {
  return <Icon name="cx-icon-trash_16x16" {...props} />;
}
