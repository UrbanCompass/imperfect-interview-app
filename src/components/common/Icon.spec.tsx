import React from 'react';
import { shallow } from 'enzyme';

import { Icon, EllipsisIcon, ExitIcon, TrashIcon } from './Icon';

describe('Icon', (): void => {
  it('render', (): void => {
    expect(shallow(<Icon name="x" left={true} />)).toMatchSnapshot();
  });

  it('renders additional class', (): void => {
    expect(shallow(<Icon name="x" right={true} className="my-icon" />)).toMatchSnapshot();
  });

  it('render ExitIcon', (): void => {
    expect(shallow(<ExitIcon />)).toMatchSnapshot();
  });

  it('render EllipsisIcon', (): void => {
    expect(shallow(<EllipsisIcon />)).toMatchSnapshot();
  });

  it('render TrashIcon', (): void => {
    expect(shallow(<TrashIcon />)).toMatchSnapshot();
  });
});
