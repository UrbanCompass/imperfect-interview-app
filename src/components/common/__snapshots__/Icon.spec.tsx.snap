// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render 1`] = `
<Styled(i)
  className="cx-icon cx-btn-icon cx-btn-icon--left"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg class=\\"cx-icon\\" viewBox=\\"0 0 24 24\\"><use xlink:href=\\"/ucfe-assets/cx-icons/4/cx-icons.cdn.svg#cx-icon-x_24x24\\"></use></svg>",
    }
  }
/>
`;

exports[`Icon render EllipsisIcon 1`] = `
<Icon
  name="cx-icon-verticalEllipsis_16x16"
/>
`;

exports[`Icon render ExitIcon 1`] = `
<Icon
  name="cx-icon-x_16x16"
/>
`;

exports[`Icon render TrashIcon 1`] = `
<Icon
  name="cx-icon-trash_16x16"
/>
`;

exports[`Icon renders additional class 1`] = `
<Styled(i)
  className="cx-icon cx-btn-icon cx-btn-icon--right my-icon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg class=\\"cx-icon\\" viewBox=\\"0 0 24 24\\"><use xlink:href=\\"/ucfe-assets/cx-icons/4/cx-icons.cdn.svg#cx-icon-x_24x24\\"></use></svg>",
    }
  }
/>
`;
